src/compile/data.ts

```diff
@@ -416,9 +506,29 @@ export namespace nullFilter {
     return nullFilterComponent;
   }
 
+  export function parseLayer(model: LayerModel) {
+    // note that we run this before source.parseLayer
+
+    // FIXME: null filters are not properly propagated right now
```

Comment:
It's not only "not properly propagated".   The merge logic below is still incorrect.  I'll add this to #1254.